src/SourceBrowser.Generator/Transformers/TreeViewTransformer.cs

```diff
@@ -34,20 +35,55 @@ public TreeViewTransformer(string savePath, string userName, string repoName)
 
         protected override void VisitWorkspace(WorkspaceModel workspaceModel)
         {
-            using (var stringWriter = new StreamWriter(_savePath, false))
-            using(_writer = new HtmlTextWriter(stringWriter))
+            // The first WorkspaceModel that is visited is the root of the tree view
+            // and its children are the solutions.
+
+            bool disposeWriters = false;
+
+            // Create the writers only if they're null
+            if (_sw == null)
             {
+                _sw = new StreamWriter(_savePath, false);
+
+                if (_writer != null)
+                    _writer.Dispose();
+
+                _writer = new HtmlTextWriter(_sw);
+
+                disposeWriters = true;
+            }
+
+            if (disposeWriters)
+            {
+                // The current WorkspaceModel is the root node, no need to increase the depth
                 _writer.AddAttribute(HtmlTextWriterAttribute.Id, "browserTree");
                 _writer.AddAttribute(HtmlTextWriterAttribute.Class, "treeview");
                 _writer.AddAttribute("data-role", "treeview");
                 _writer.RenderBeginTag(HtmlTextWriterTag.Ul);
-
+            }
+            else
+            {
+                // The current WorkspaceModel is a Child of the root node.
                 depth++;
-                base.VisitWorkspace(workspaceModel);
-                depth--;
+            }
+            
+            base.VisitWorkspace(workspaceModel);
 
+            if (disposeWriters)
+            {
+                // The current WorkspaceModel is the root node.
+                // Every child has been visited: dispose the writers.
+
+                disposeWriters = false;
                 _writer.RenderEndTag();
                 _writer.WriteLine();
+                _writer.Dispose();
+                _sw.Dispose();
```

Comment:
Hi,
I tried in implementing the changes that I described above, but in some cases there are problems with repositories with many solutions such that the result is not really good.
Take for example [Hangfire](https://github.com/HangfireIO/Hangfire): there are two solutions in the root and they share some source files. By flattening the hierarchy in the root WorkplaceModel, the generated treeview is not really readable

<img width="242" alt="sb_hang_old" src="https://cloud.githubusercontent.com/assets/8939890/24324652/a65a4076-118a-11e7-851e-8d466d11ac3e.PNG" />

I tried to solve this by adding the solutions' children to a new `FolderModel` instead of directly to the root, and indeed the new treeview was better

<img width="243" alt="sb_hang_new" src="https://cloud.githubusercontent.com/assets/8939890/24324710/96f529b0-118b-11e7-95ee-d8319097c145.PNG"/>

but then the navigation didn't work well because the relative paths didn't match the files in the treeview anymore.

I'm thinking that this could be solved by navigating to a solution's treeview when you click on the solution's name, so basically when you go to */Browse/{user}/{repo}* the treeview section shows the available solutions.

But before going this path, I need your opinions.
Thank you